File: 2. Managers/MusicAPI.Managers/ViewModels/Artist.cs
namespace MusicAPI.Managers.ViewModels
{
    public class Artist
    {
        public ExternalUrl ExternalUrl { get; set; } = new ExternalUrl();

        public Followers Followers { get; set; } = new Followers();

        public string[] Genres { get; set; } = [];

        public string Href { get; set; } = string.Empty;

        public string Id { get; set; } = string.Empty;

        public Image[] Images { get; set; } = [];

        public string Name { get; set; } = string.Empty;

        public int Popularity { get; set; } = 0;

        public string Type { get; set; } = string.Empty;

        public string Uri { get; set; } = string.Empty;
    }
}

File: 1. Clients/angular-showcase/src/app/modules/data-models/artist.ts
import { ExternalUrl } from "./externalUrl";
import { Followers } from "./followers";
import { Image } from "./image";

export class Artist {
  constructor() {
    this.externalUrl = new ExternalUrl();
    this.followers = new Followers();
    this.genres = [];
    this.href = '';
    this.id = '';
    this.images = [];
    this.name = '';
    this.popularity = 0;
    this.type = '';;
    this.uri = '';
  }

  externalUrl: ExternalUrl;
  followers: Followers;
  genres: string[];
  href: string;
  id: string;
  images: Image[];
  name: string;
  popularity: number;
  type: string;
  uri: string;
}
